lib/miq_automation_engine/engine/drb_remote_invoker.rb

```diff
@@ -63,10 +63,15 @@ def drb_cache_timeout
       1.hour
     end
 
+    def api_token
+      tm = TokenManager.new("api")
+      tm.gen_token("api", :userid => @workspace.ae_user.userid)
+    end
+
     # code building
 
-    def method_preamble(miq_uri, miq_id)
-      "MIQ_URI = '#{miq_uri}'\nMIQ_ID = #{miq_id}\n" << RUBY_METHOD_PREAMBLE
+    def method_preamble(miq_uri, miq_id, miq_api_token)
+      "MIQ_URI = '#{miq_uri}'\nMIQ_ID = #{miq_id}\nMIQ_API_TOKEN = '#{miq_api_token}'\n" << RUBY_METHOD_PREAMBLE
```

Comment:
I guess that depends on the time span of these method invocations.  Default TTL's for api tokens is 10 minutes (they auto-reset upon use).